index.js

```diff
@@ -255,12 +260,13 @@ class ClientServiceBase {
         ],
     );
 
-    const signerFactory = new SignerFactory(
+    this.signer = this.signer || this.signerFactory.create(
```

Comment:
Seeing this I understand what you meant by the `lazy loading` you said during the meeting earlier. 
I think it's a good idea but to avoid duplicating in each method 
```
this.signer = this.signer || this.signerFactory.create(
        clientProperties.getPrivateKeyCryptoKey() ||
        clientProperties.getPrivateKeyPem()
    );
```
Could a signer getter method be created (I am not sure about the syntax or naming)
```
function _getSigner() {
this.signer = this.signer || this.signerFactory.create(
        clientProperties.getPrivateKeyCryptoKey() ||
        clientProperties.getPrivateKeyPem()
    );
return this.signer;
}
```
Then you would just have to call this method in each request method instead ?

Reply:
It's a good idea. In case we modify the code later.